src/cc/common/buildversgit.sh

```diff
@@ -20,60 +11,178 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
 # implied. See the License for the specific language governing
 # permissions and limitations under the License.
-#
-# Create Version.cc file with the relevant source / build information.
-#
-#
 
-if [ x"$1" = x'-g' -o  x"$1" = x'-get' ]; then
-    while [ $# -gt 1 ]; do
-        shift
-        strings -a "$1" | awk '/KFS_BUILD_INFO_START/,/KFS_BUILD_INFO_END/'
-    done
-    exit
-fi
+usage() {
+    echo "
+usage: $0 <options>
+
+    Options:
+        --info BINARY               get build information from the binary
+        --head                      print the head hash of qfs
+        --release                   print the release verison or hash of qfs
+        --remote                    print the remote we are tracking
+        --branch                    print the branch we are on
+
+        --generate                  generate build info -- if this option is
+                                    specified, then --build-type, --source-dir,
+                                    and --outfile are all required
+        --build-type BUILD_TYPE     the build type we built with
+        --source-dir SOURCE_DIR     the directory where qfs source lives
+        --outfile OUTFILE           the file to write generate c++ code
+        --extra EXTRA               extra information to include in the output
+                                    this parameter can be used multiple times
+
+        --help                      show this help
+    "
+    exit 1
+}
 
-# Set official release version here.
-qfs_release_version="1.1.4"
-qfs_source_revision=""
+get_head() {
+    if ! git="$(command -v git)" || [ -z "$git" ]; then
+        echo "unknown"
+        return 0
+    fi
 
```

Comment:
Done. Thanks! It actually adds a bit more lines but it encapsulates _how_ we check for git presence.